packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js

```diff
@@ -84,107 +102,178 @@ renderHeaderItem.propTypes = PropTypes.oneOfType([
 	])),
 ]);
 
-function renderHeader({ header, caret }) {
+function renderHeader(header, content, onSelect, onToggle) {
 	const headerColumnClass = `col-${header.length}`;
 	const headerItems = header.map((headerItem, index) => {
 		if (Array.isArray(headerItem)) {
 			const elements = headerItem.map(renderHeaderItem);
 			return (
-				<div key={index} className={classNames(css.group, css[headerColumnClass])}>{elements}</div>
+				<div
+					key={index}
+					className={classNames(css.group, css[headerColumnClass])}
+				>{elements}</div>
 			);
 		}
 		return (
-			<div key={index} className={css[headerColumnClass]}>{renderHeaderItem(headerItem)}</div>
+			<div
+				key={index}
+				className={classNames(css[headerItem.className], css[headerColumnClass])}
+			>{renderHeaderItem(headerItem)}</div>
 		);
 	});
 
-	if (caret) {
-		headerItems.push(
-			<Icon
-				key={header.length}
-				className={css.caret}
-				name={'talend-caret-down'}
-			/>,
+	// Panel component needs an array for header props
+	const wrappedHeader = [
+		onSelect ? (
+			<Button
+				className={css['left-side']}
+				bsStyle="link"
+				key={1}
+				onClick={onSelect}
+			>
+				<div className={classNames(css['panel-title'], 'panel-title')}>
+					{headerItems}
+				</div>
+			</Button>
+		) : (
+			<div
+				className={css['left-side']}
+				bsStyle="link"
+				key={1}
+			>
+				<div className={classNames(css['panel-title'], 'panel-title')}>
+					{headerItems}
+				</div>
+			</div>
+		),
+	];
+
+	if (content) {
+		const defaultCaret = (
+			<Button
+				className={css.toggle}
+				bsStyle="link"
+				key={2}
+				onClick={onToggle}
+			>
+				<Icon
+					key={header.length}
+					name="talend-caret-down"
+				/>
+			</Button>
 		);
+		wrappedHeader.push(defaultCaret);
 	}
+	return wrappedHeader;
+}
+
+function getKeyValueContent(content) {
+	return (<dl>
+		{content.map(
+			(item, index) => (
+				<div key={index} className={css.content}>
+					<dt className={css.label}>
+						<Label>{item.label}</Label>
+					</dt>
+					<dd className={css.description}>
+						{item.description}
+					</dd>
+				</div>)
+		)}
+	</dl>);
+}
 
-	return headerItems;
+function getTextualContent(content) {
+	return (
+		<div className={css.content}>
+			<div className={css.upper}>
+				{content.upper.map((item, index) => {
+					if (typeof item === 'string') {
+						return (<span key={index}>{item}</span>);
+					}
+					const { label, tooltipPlacement } = item;
+					return (
+						<TooltipTrigger
+							key={index}
+							label={label}
+							tooltipPlacement={tooltipPlacement}
+						>
+							<span>{label}</span>
+						</TooltipTrigger>
+					);
+				})}
+			</div>
+			<div className={css['content-description']}>
```

Comment:
Keep in mind that we may want to add some style in our final apps, you should expose not computed class name